### PR TITLE
[MOBILE-3858] Add support for feature flag

### PR DESCRIPTION
**Pull request by Ulrico972**
### What do these changes do?
Add support for feature flag

### Why are these changes necessary?
To support feature flag

### How did you verify these changes?
Manual testing

### CODE REVIEW

File: ios/Classes/SwiftAirshipPlugin.swift [hunk 1]
```diff
@@ -405,6 +405,12 @@ public class SwiftAirshipPlugin: NSObject, FlutterPlugin {
             ) as? AirshipJSON
             return result?.unWrap()
 
+        // Feature Flag
+        case "featureFlag#flag":
+            return try await AirshipProxy.shared.featureFlagManager.flag(
+                name: try call.requireStringArg()
+            )
```
Comment (rlepinski) [inline ios/Classes/SwiftAirshipPlugin.swift:412]: I think we need to jsonify this result. `JSONValue.wrapOpt(result).unwrap` or whatever

File: lib/src/feature_flag.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  /// Indicates whether the flag exists in the current flag listing or not.
+  final bool exists;
+
+  /// Optional variables associated with the flag.
+  final Map<String, dynamic>? variables;
+
+  const FeatureFlag._internal(this.isEligible, this.exists, this.variables);
+
+  static FeatureFlag fromJson(Map<String, dynamic> json) {
```
Comment (rlepinski) [inline lib/src/feature_flag.dart:16]: pass in dynamic here

File: lib/src/feature_flag.dart [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    if (json == null) {
+      return null;
+    }
+
+    try {
+      var isEligible = json["isEligible"];
+      var exists = json["exists"];
+      var variables = Map<String, dynamic>.from(json["variables"]);
+      return FeatureFlag._internal(isEligible, exists, variables);
+    } catch (e, s) {
```
Comment (rlepinski) [inline lib/src/feature_flag.dart:26]: I think we should just let it crash?

File: lib/src/feature_flag.dart [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  /// Indicates whether the flag exists in the current flag listing or not.
+  final bool exists;
+
+  /// Optional variables associated with the flag.
+  final Map<String, dynamic>? variables;
+
+  const FeatureFlag._internal(this.isEligible, this.exists, this.variables);
+
+  static FeatureFlag? fromJson(Map<String, dynamic>? json) {
```
Comment (rlepinski) [inline lib/src/feature_flag.dart:16]: I would make this ` static FeatureFlag fromJson(dynamic json)`

**Comment by rlepinski** [comment]
I updated it with my feedback since ill be out. Here is the usage:
```
Airship.featureFlagManager.flag("another_rad_flag").then((value) => {
    debugPrint("flag: $value")
}).catchError((error) => {
    debugPrint("flag error: $error")
});
```

**Comment by Ulrico972** [comment]
We have an issue on iOS, the proxy has a typo in one feature flag key. I think ryan found it but I'm not sure how he planed to handle it.
The best way would be to fix the proxy but it might be used already in other frameworks. So I'm checking them to see what's best. Maybe I'll find the workaround in one of them.

**Review by jyaganeh — APPROVED** [review]

**Pull request merged**